Main.hs

```diff
@@ -4,22 +4,48 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short (_ : _ : _ : _ ) = False
+short _ = True
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely l
+    | short(l) = True
+    | l!!2 == 14 = True
+    | otherwise = False
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [(x, y, z) |z <- [1..], y <- [1..z], x <- [1..y], x * x + y * y == z * z]
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [logic x | x <- [1..]]
+    where
+        logic x  
+            | x `mod` 15 == 0 = "FizzBuzz"
+            | x `mod` 5 == 0 = "Buzz"
+            | x `mod` 3 == 0 = "Fizz"
+            | otherwise = show(x)
```

Comment:
Avoid using unnecessary parentheses: Haskell is not Java, you don't need to put arguments of a function in parentheses.